### ISSUE TITLE
Error: '"Must have equal len keys and value when setting with an iterable"

### ISSUE THREAD

**Issue by khamamoto6** [Sinaptik-AI/pandas-ai#353]
### 🐛 Describe the bug

PandasAI seems to be unable to deal with data frames when there are columns of list elements, returning the error "Must have equal len keys and value when setting with an iterable". Please see example below: 

```
import pandas as pd
from pandasai import PandasAI

# Sample DataFrame
df = pd.DataFrame({
    "house": ["a", "b", "c"],
    "happiness_index": [6.94, 7.16, 6.66],
	"people": [['abby', 'ben'], ['ken'], []]
})

# Instantiate a LLM
from pandasai.llm.openai import OpenAI
llm = OpenAI()

pandas_ai = PandasAI(llm)
pandas_ai.run(df, prompt='Which house has at least two people?')
```

This returns that error I mentioned above.

**Comment by nautics889** [comment]
Hello, @khamamoto6
___
The problem arises from [anonymizer.py](https://github.com/gventuri/pandas-ai/blob/main/pandasai/helpers/anonymizer.py) module, when pandasai [tries to anonymize dataframe's head](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L360).
To be accurate, it fails when trying to set a dataframe's unit having `random_value` equal to `[]` (an empty list). The pandas' `_iLocIndexer` can't set item value to empty list for some reason. For now i couldn't figure out a quick solution for the issue. However, if you set something like this in your source data:
```
df = pd.DataFrame({
    "house": ["a", "b", "c"],
    "happiness_index": [6.94, 7.16, 6.66],
	"people": [['abby', 'ben'], ['ken'], ['ian']]
})
```
it will not fail with that error :) 
Although there is still some different problem — an empty string in the result (running with code given above), i'm going to describe this in the separate issue.

**Comment by nautics889** [comment]
It should have been fixed in [PR #372](https://github.com/gventuri/pandas-ai/pull/372); that's exactly about the problem which i've described [above](https://github.com/gventuri/pandas-ai/issues/353#issuecomment-1634812989).
___
@gventuri should the current be closed as well?=)
